packages/block-editor/src/components/block-list/block.js

```diff
@@ -105,13 +106,14 @@ function BlockListBlock( {
 	const wrapper = useRef( null );
 	useEffect( () => {
 		blockRef( wrapper.current, clientId );
-		// We need to rerender to trigger a rerendering of HoverArea.
-		rerender();
```

Comment:
I guess this is where we get most of the performance benefit? We're currently rendering twice for every component update?

Reply:
yes, I think in my tests, the difference is not really noticeable but yeah it's way cleaner that way.

Reply:
To clarify we're not rerendering on every update but we're rendering twice on mount.

Reply:
It might be easier to reason about with something like this:
https://github.com/streamich/react-use/blob/master/docs/useMount.md
?
shall I add this utility function ? not the whole library

Reply:
I read in several places that ideally when using hooks we should try to avoid thinking in terms of lifecycle hooks (did mount, did update) and instead thinking in terms of effects as results of props changes, state changes... Makes me wonder if we want to integrate this kind of hooks.